README.md

```diff
@@ -1 +1,110 @@
-A simple CSV reader based on processing chunks of input.
\ No newline at end of file
+# ChunkedCSV.jl
+
+The main goal of this package is to offer a parser for CSV files that can be used for streaming purposes. This means that it can work with data that is too large to fit into memory or in situations where we only need to process parts of the input data at a time. The parser reads the data in chunks, and it can parse each chunk in parallel while simultaneously prefetching the next one to increase throughput. Additionally, the parsed data is passed to a user-defined function that can consume it in any way required.
+
+Moreover, the package can automatically handle [`Parsers.jl`](https://github.com/JuliaData/Parsers.jl)-compatible types without dynamic dispatch overhead and with a low compilation overhead. This is done without specializing on input schema, and the memory usage is upper bound by the user, who can control the chunk size. All internally allocated buffers are reused, and the memory overhead for inputs with uniform density is fixed.
```

Comment:
This was supposed to say that if there is roughly the same number of values per MiB of input, then we don't have to allocate since we're reusing buffers for parsed values and newlines between each chunk. But if the number of values per MiB of input increases (e.g. at the start of the file, all columns are densely populated, and by the end of the file, there are mostly missing values / empty fields), then we'd need to grow out buffers to handle the increasing number of values per MiB of input.